Lecture4/src/main/java/Validator.java

```diff
@@ -0,0 +1,58 @@
+import java.util.InputMismatchException;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Validator {
+	private double operand;
+	private Operator operator;
+	private static Scanner input = new Scanner(System.in);
+	private static final Pattern NUMBER_PATTERN = Pattern.compile("-?[0-9]\\d*(\\.\\d+)?");
+
+	public double getOperand() {
+		return operand;
+	}
+
+	public Operator getOperator() {
+		return operator;
+	}
+
+	public boolean isNumber() {
```

Comment:
you don't use result of your methods
i suggest to rename it to ` int getNumbers()`  and return result to main class - and use it there